packages/gatsby-theme-docs/src/layouts/internals/sidebar.js

```diff
@@ -91,55 +95,151 @@ const LogoTitle = styled.div`
   }
 `;
 
-const SidebarLink = props => (
-  <ClassNames>
-    {({ css: makeClassName }) => {
-      const linkClassName = makeClassName`
-        border-left: ${designSystem.dimensions.spacings.xs} solid
-          ${designSystem.colors.light.surfacePrimary};
-        padding-left: calc(
-          ${designSystem.dimensions.spacings.m} - ${designSystem.dimensions.spacings.xs}
-        );
-        text-decoration: none;
-        color: ${designSystem.colors.light.textSecondary};
-        display: flex;
-        flex-direction: row;
-        align-items: flex-end;
+const SidebarLink = React.forwardRef((props, ref) => {
+  // Filter out props that we don't want to forward to the Link component
+  const { location, nextScrollPosition, ...forwardProps } = props;
 
-        :hover {
-          color: ${designSystem.colors.light.linkNavigation} !important;
-        }
+  const cachedScrollPosition = (location.state || {}).scrollPosition;
+  const locationPath = trimTrailingSlash(location.pathname);
 
-        > * + * {
-          margin: 0 0 0 ${designSystem.dimensions.spacings.s};
-        }
-      `;
-      const activeClassName = makeClassName`
-        border-left: ${designSystem.dimensions.spacings.xs} solid ${designSystem.colors.light.linkNavigation} !important;
-        color: ${designSystem.colors.light.linkNavigation} !important;
-      `;
-      return (
-        <Link
-          {...props}
-          // eslint-disable-next-line react/prop-types
-          to={trimTrailingSlash(props.to)}
-          getProps={({ href, location }) => {
-            // Manually check that the link is the active one, even with trailing slashes.
-            // The gatsby link is by default configured to match the exact path, therefore we
-            // need to check this manually.
-            const linkPath = trimTrailingSlash(href);
-            const locationPath = trimTrailingSlash(location.pathname);
-            if (linkPath === locationPath) {
-              return { className: [linkClassName, activeClassName].join(' ') };
-            }
-            return { className: linkClassName };
-          }}
-        />
-      );
-    }}
-  </ClassNames>
-);
+  const linkRef = React.useRef();
+  const scrollIntoView = React.useCallback(() => {
+    ref.current.scrollIntoView({ block: 'center' });
```

Comment:
I tried start but as you mentioned it cold be confusing because you potentially don’t the above chapters. I can change it though if you prefer it like this, I don’t have strong opinions on this.

Reply:
I tried with some manual scripting and actually liked the behavior although I initially thought it could be confusing.